src/context/Confirmation.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+} from '@mui/material';
+import React, { createContext, useContext, useRef, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+export interface IConfirmationModalOptions {
+    confirmText?: string;
+    cancelText?: string;
+    title?: string;
+    message: string;
+}
+
+interface IConfirmationModalContext {
+    showConfirmation: ({
+        title,
+        message,
+        confirmText,
+        cancelText,
+    }: IConfirmationModalOptions) => Promise<any>;
+}
+
+const getDefaultSettings = (): IConfirmationModalOptions => {
+    return {
+        title: 'confirm.title',
+        message: '',
+        confirmText: 'cta.continue',
+        cancelText: 'cta.cancel',
+    };
+};
+
+const ConfirmationModalContext =
+    createContext<IConfirmationModalContext | null>(null);
+
+const ConfirmationModalContextProvider: React.FC = ({ children }) => {
+    const [settings, setSettings] = useState<IConfirmationModalOptions>(
+        getDefaultSettings()
+    );
+    const [showConfirmationModal, setShowConfirmationModal] = useState(false);
+    const resolver = useRef<any>();
+
+    const handlers = {
+        confirm: () => {
+            resolver.current?.(true);
+            setShowConfirmationModal(false);
+        },
+        dismiss: () => {
+            resolver.current?.(false);
+            setShowConfirmationModal(false);
+        },
+        show: (userSettings: IConfirmationModalOptions) => {
+            setSettings({
+                ...getDefaultSettings(),
+                ...userSettings,
+            });
+            setShowConfirmationModal(true);
+
+            return new Promise((resolve) => {
+                resolver.current = resolve;
+            });
+        },
```

Comment:
We set the settings at the time the confirmation is displayed. I don't think this would be the best way to handle this if this was used in multiple places but we should never have multiple confirmation dialogs at once.